velox/vector/tests/VectorTest.cpp

```diff
@@ -1542,6 +1542,88 @@ TEST_F(VectorTest, wrapInConstantWithCopy) {
   }
 }
 
+TEST_F(VectorTest, rowResize) {
+  auto testRowResize = [&](std::vector<VectorPtr> children, bool setNotNull) {
+    auto rowVector = vectorMaker_.rowVector(children);
+    auto oldSize = rowVector->size();
+    auto newSize = oldSize * 2;
+
+    rowVector->resize(newSize, setNotNull);
+
+    EXPECT_EQ(rowVector->size(), newSize);
+
+    for (auto& child : children) {
+      EXPECT_EQ(child->size(), newSize);
+    }
+
+    if (setNotNull) {
```

Comment:
No if setNotNull is false, the bits from oldSize to newSize can be anything (true or false) , setNotNull to true guarantees its true, but setting it to false doesnt mean it will set it all to false. This is an optimization.